### ISSUE TITLE
Errors thrown inside setTimeout lose their stack trace

### ISSUE THREAD

**Issue by matt-smarsh** [vitest-dev/vitest#1779]
### Describe the bug

Errors thrown in a `setTimeout()` block do not receive a stack trace when running in Node v16.16.0

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-m7mxed?file=test%2Fbasic.test.ts

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (12) x64 Intel(R) Core(TM) i7-10750H CPU @ 2.60GHz
    Memory: 3.98 GB / 15.75 GB
  Binaries:
    Node: 16.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 8.14.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (103.0.1264.77)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    vite: 3.0.4 => 3.0.4
    vitest: 0.20.3 => 0.20.3
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by matt-smarsh** [comment]
Output from Node v16.16.0:
![image](https://user-images.githubusercontent.com/89040187/182719533-62d63ab2-d405-47aa-9804-cc31f3755aa0.png)

**Comment by sheremet-va** [comment]
Fixed in #1779
